content/english/blog/dotnet_are_we_webassembly_yet.md

```diff
@@ -0,0 +1,115 @@
+image : "images/blogs/dotnet_wasm.png"
+date: 2022-11-12T9:00:00-04:00
+author: "Kevin Hoffman"
+author_profile: "https://www.linkedin.com/in/%F0%9F%A6%80-kevin-hoffman-9252669/"
+description : "Checking back in on the .NET+WebAssembly ecosystem"
+categories: ["webassembly", "dotnet", ".net", "blazor", "csharp", "wasi"]
+draft : false
+---
+
+With the recent [release of the .NET Framwework 7](https://devblogs.microsoft.com/dotnet/announcing-dotnet-7/), I thought it might be a good excuse to check back in on the .NET WebAssembly ecosystem and see where things stand and what improvements have been made. If you want to follow along or experiment on your own, the first thing you'll want to do is [install .NET 7](https://dotnet.microsoft.com/en-us/download/dotnet/7.0).
+
+In nearly every case, whenever I talk to anyone about .NET and WebAssembly, the conversation shifts quickly to [Blazor](https://dotnet.microsoft.com/en-us/apps/aspnet/web-apps/blazor). Blazor, in a vacuum, doesn't interest me. Not because it isn't good, but because _Blazor owns both ends of the pipeline: the producer and the consumer_.
+
+The practical impact of this is that they can pretty much do whatever they like. They are free of the burdens and constraints that inhibit most of us trying to work in the cloud native/backend space with WebAssembly. To see what this looks like and so you can _feel_ the smoke and mirrors at work, let's take a look at what .NET 7 thinks of as a "pure" WebAssembly application.
+
+Type `dotnet new list` and you'll see all the templates available for you to create things:
+
+```
+These templates matched your input:
+
+Template Name                       Short Name          Language    Tags
+----------------------------------  ------------------  ----------  --------------------------------
+ASP.NET Core Empty                  web                 [C#],F#     Web/Empty
+ASP.NET Core gRPC Service           grpc                [C#]        Web/gRPC
+...
+Blazor Server App                   blazorserver        [C#]        Web/Blazor
+Blazor Server App Empty             blazorserver-empty  [C#]        Web/Blazor/Empty
+Blazor WebAssembly App              blazorwasm          [C#]        Web/Blazor/WebAssembly/PWA
+Blazor WebAssembly App Empty        blazorwasm-empty    [C#]        Web/Blazor/WebAssembly/PWA/Empty
+...
+WebAssembly Browser App             wasmbrowser         [C#]        Web/WebAssembly/Browser
+WebAssembly Console App             wasmconsole         [C#]        Web/WebAssembly/Console
+...
+```
+I've trimmed out the templates that don't apply to this discussion. I was so excited when I saw the `wasmconsole` project template, I grabbed some coffee, dropped what I was doing, and started tinkering. _"Finally," I thought. "Freestanding WASM support"_.
+
+Let's create a new wasmconsole via ``dotnet new wasmconsole`` (note that unlike Rust, this creates the project in your current directory, not a sub-directory).
+
+```
```

Comment:
I'd suggest adding language annotations like `terminal` or `shell` here so that the backticked sections are distinguished

Reply:
when I add those, the rendering is completely illegible - dark red on black with 3 blanks per line.

Reply:
Gross. Disregard then!